### PR TITLE
SQL: Simplify function registration and resolution

### PR DESCRIPTION
**Pull request by matriv**
Previously, we used a CamelCase to CAMEL_CASE transformation to get the
primary name of a function from its class name which led to some issues
since there are functions that we don't want to be registered this way
(e.g.: IFNULL). To simplify the logic and avoid and "magic"
transformations in the FunctionRegistry a primary name must be provided
explicitely for each function.

The same change is applied for the function resolution (when a function
is used in an SQL statement). There is no CamelCase to CAMEL_CASE
transformation but only upper-casing is applied (fuNcTiOn -> FUNCTION).

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/FunctionRegistry.java [hunk 1]
```diff
@@ -454,8 +444,13 @@ public boolean functionExists(String functionName) {
 
     @SuppressWarnings("overloads")
     private static FunctionDefinition def(Class<? extends Function> function, FunctionBuilder builder,
-            boolean datetime, String... aliases) {
-        String primaryName = normalize(function.getSimpleName());
+            boolean datetime, String... names) {
+        Check.isTrue(names.length > 0, "At least one name must be provided for the function");
+        String primaryName = names[0];
+        List<String> aliases = new ArrayList<>(names.length - 1);
```
Comment (astefan) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/FunctionRegistry.java]: I would add a comment here or a javadoc for the method stating that the first name in the array is the actual name of the function and the rest (if any) are aliases.

**Review by astefan — APPROVED** [review]
LGTM. Left one comment.

**Review by costin — APPROVED** [review]
LGTM!

**Comment by matriv** [comment]
Backported to `6.x` with 2a811725948500cf7b3cf4e22ab1647df2e6286b

**Pull request merged**